CRM/Core/Payment/Stripe.php

```diff
@@ -837,4 +837,16 @@ public function doRecurPayment(&$params, $amount, $stripe_customer) {
   public function doTransferCheckout(&$params, $component) {
     CRM_Core_Error::fatal(ts('Use direct billing instead of Transfer method.'));
   }
+
+  /**
+   * Process incoming notification.
+   */
+
+  static public function handlePaymentNotification() {
+    $data_raw = file_get_contents("php://input");
```

Comment:
I don't like this line... Most of the payment processors seem to do something like `$params = array_merge($_GET, $_REQUEST);`

Can we do that here?

Reply:
That format also caught my eye because I'm not used to that format either.

However, I investigated and noticed that the Stripe payment processor recommends that syntax (https://stripe.com/docs/webhooks) and the PHP manual suggests that in some cases it is better then $_POST (https://secure.php.net/manual/en/wrappers.php.php). 

But most importantly, it will fail if we switch to $_POST because stripe is not sending a normal key value pair - instead it is sending raw JSON.

Here's the difference - I wrote a test script:

```
<?php
print_r($_POST);
print_r(file_get_contents("php://input")); 
echo "\n";
```        
And here's the difference:
```
0 jamie@turkey:cdtemp.Q7dESJ$ curl --data-raw '{"name":"bob"}' https://wfan.loc.cx/test.php
Array
(
    [{"name":"bob"}] => 
)
{"name":"bob"}
0 jamie@turkey:cdtemp.Q7dESJ$ 
```
If we use $_POST we have to extract the index which is cumbersome compared with simply getting the raw data, which is what we ultimately want.